tx-pool/src/component/pending.rs

```diff
@@ -84,7 +84,7 @@ impl CellProvider for PendingQueue {
                     let mut cell_meta = CellMetaBuilder::from_cell_output(output, data)
                         .out_point(out_point.to_owned())
                         .build();
-                    if !with_data {
+                    if !allow_in_txpool && !with_data {
```

Comment:
This line looks like changed the original behavior, I think the original code is wrong, should be:
```
                    if !with_data {
                        cell_meta.mem_cell_data_hash = None;
                    }
``